### PR TITLE
Update readme with more information

### PR DESCRIPTION
**Pull request by neurosnap**
I think a **Why** section would be useful for the documentation because it would explain the benefits of not using JSX with react.

An **API** section helped me understand all the different configurations in a clear and concise manner.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,60 @@
...
+## Why?
+
+### Pros
+* Consistent javascript syntax
+* Mistyped components return errors
+* No need to litter code with `null` values when a component doesn't have any props
+* No need for a JSX syntax highlighter
+* No need for a JSX linter
+* JSX elements are just functions anyway, why try to hide what it is?
+* HTML markup is ugly
```
Comment (Jador) [inline README.md]: too subjective

**Comment by Jador** [comment]
Thanks a lot for this, especially the api section. There was one little thing I took exception with, which I commented on. But then I'll be happy to accept this PR.

**Comment by Jador** [comment]
I just noticed that most of this comes from the readme for your project, which is perfectly fine with me. I would like it if you brought over the alternatives section too.

**Comment by neurosnap** [comment]
My project was more of a way for me to figure out how I would approach dealing with `React.createElement`, I don't really plan on publishing it because it's virtually identical to this library.

**Comment by neurosnap** [comment]
What part of the Why section is too subjective?  All of it or just the comment about HTML markup being ugly?

**Comment by Jador** [comment]
The part about the HTML (not that I don't agree lol)

**Comment by neurosnap** [comment]
I updated the **Pros** section to remove the subjective comments.

I also added an **Alternatives** section.

What do you think about coming up with some examples to illustrate the differences between JSX, `React.createElement`, and rhh in the README?

**Comment by Jador** [comment]
Sure, sounds like a good idea

**Comment by Jador** [comment]
@neurosnap, I know if you're like me you probably don't have a ton of time to devote to this. If you like I can go ahead and accept this PR now and you can make another for the api differences later.

**Comment by neurosnap** [comment]
I'd prefer to send a separate PR for that, but I am working on it.  Any suggestions for a good example?  Right now I'm using redux todo as my template.

**Pull request merged**